integrations/benchmarks_test.go

```diff
@@ -74,40 +77,48 @@ func StringWithCharset(length int, charset string) string {
 }
 
 func BenchmarkRepoBranchCommit(b *testing.B) {
-	samples := []int64{1, 3, 15, 16}
-	defer prepareTestEnv(b)()
-	b.ResetTimer()
+	onGiteaRunTB(b, func(t testing.TB, u *url.URL) {
+		b := t.(*testing.B)
 
-	for _, repoID := range samples {
-		b.StopTimer()
-		repo := models.AssertExistsAndLoadBean(b, &models.Repository{ID: repoID}).(*models.Repository)
-		b.StartTimer()
-		b.Run(repo.Name, func(b *testing.B) {
-			owner := models.AssertExistsAndLoadBean(b, &models.User{ID: repo.OwnerID}).(*models.User)
-			session := loginUser(b, owner.LoginName)
-			b.ResetTimer()
-			b.Run("Create", func(b *testing.B) {
-				for i := 0; i < b.N; i++ {
+		samples := []int64{1, 2, 3}
+		b.ResetTimer()
+
+		for _, repoID := range samples {
+			b.StopTimer()
+			repo := models.AssertExistsAndLoadBean(b, &models.Repository{ID: repoID}).(*models.Repository)
+			b.StartTimer()
+			b.Run(repo.Name, func(b *testing.B) {
+				session := loginUser(b, "user2")
+				b.ResetTimer()
+				b.Run("CreateBranch", func(b *testing.B) {
+					b.Skip("benchmark broken") // TODO fix
 					b.StopTimer()
-					branchName := StringWithCharset(5+rand.Intn(10), "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
+					branchName := "new_" + StringWithCharset(5+rand.Intn(10), "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789")
```

Comment:
for some reason lint doesn't like this, it thinks `StringWithCharset` is unused 🤔